File: java/src/test/java/TransformTest.java
import org.junit.jupiter.api.Test;

import java.io.IOException;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class TransformTest {

  @Test
  public void testMenuTransform() throws IOException {
    final Map<RestaurantDishKey, CaloriesVegan> map = TransformStream.run();

    final CaloriesVegan actual = map.get(new RestaurantDishKey("Cheval Blanc", "Risotto"));
    assertEquals(450, actual.getCalories());
    assertTrue(actual.isVegan());
  }
}

File: kotlin/src/test/kotlin/TransformTest.kt
import org.junit.jupiter.api.Test

import java.io.IOException
import java.nio.file.Paths
import kotlin.test.assertEquals
import kotlin.test.assertTrue

class TransformTest {

  @Test
  @Throws(IOException::class)
  fun testMenuTransform() {

    val map = transform(Paths.get("../restaurants.json"))

    map[RestaurantDishKey("Cheval Blanc", "Risotto")]!!.apply {
      assertEquals(450, calories)
      assertTrue(isVegan)
    }
  }
}
